lib/promscrape/targetstatus.go

```diff
@@ -355,6 +360,25 @@ func (tsm *targetStatusMap) getTargetsStatusByJob() ([]jobTargetsStatuses, []str
 	return jts, emptyJobs
 }
 
+func filterJobs(jts []jobTargetsStatuses, searchQuery string) []jobTargetsStatuses {
+	regQuery := filterReg.ReplaceAllString(searchQuery, ")(?=.*")
```

Comment:
It would be great if `searchQuery` could contain arbitrary [time series selector](https://prometheus.io/docs/prometheus/latest/querying/basics/#time-series-selectors). For example `{job=~"foo.*"}` would match all the targets with `job` label, which starts from `foo`.
I suppose the code from [lib/promrelabel/if_expression.go](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/60ad8c74bccad658a08bd038ea93d0c8efaa3abd/lib/promrelabel/if_expression.go#L49) could be re-used for this case.

Reply:
Used this library and split logic into two separate functions